test/dynamo/test_aot_autograd.py

```diff
@@ -1190,6 +1195,133 @@ def fn(x):
         with self.assertRaises(Exception):
             opt_fn(x_opt)
 
+    @torch._functorch.config.patch(donated_buffer=True)
+    def test_donated_buffer1(self):
+        logger_name = "torch._functorch._aot_autograd.jit_compile_runtime_wrappers"
+
+        @torch.compile()
+        def relu(x):
+            return torch.nn.functional.relu(x)
+
+        with self.assertLogs(logger_name, level="INFO") as captured:
+            relu(torch.rand([3, 3], requires_grad=True)).sum().backward()
+
+        if is_dynamic_shape_test(self._testMethodName):
+            # an extra symint exists
+            expected_msg = "bw_donated_idxs=[1]"
+        else:
+            expected_msg = "bw_donated_idxs=[0]"
+
+        # le is a donated buffer from relu
+        FileCheck().check(expected_msg).run("\n".join(captured.output))
```

Comment:
Hmm... I don't think we have many tests that grab logs and assert on their output. I guess this one is ok since we shouldn't be wobbling the AOTAutograd logs very much (that log ViewAndMutationMeta).

Just to give an alternative, you could write a testing compile backend similar to inductor that calls `aot_autograd(fw_compiler, bw_compiler)`, where `fw_compiler()` grabs `TracingContext().fw_metadata` and asserts on the `bw_donated_idxs` directly (e.g. [here](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/compile_fx.py#L1251))